src/Workspaces/Core/Portable/EmbeddedLanguages/DateTime/LanguageServices/DateTimeEmbeddedLanguage.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification.Classifiers;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.DateTime.LanguageServices
+{
+    internal class DateTimeEmbeddedLanguage : IEmbeddedLanguage
+    {
+        public readonly EmbeddedLanguageInfo Info;
+
+        public ISyntaxClassifier Classifier { get; }
+
+        public DateTimeEmbeddedLanguage(EmbeddedLanguageInfo info)
+        {
+            Info = info;
+            Classifier = new FallbackSyntaxClassifier(info);
+        }
+
+        internal async Task<SyntaxToken?> TryGetDateTimeStringTokenAtPositionAsync(
```

Comment:
Curious why this needs to be a separate class rather than part of the completion provider?  seems to be only used once

Reply:
i'm following the regex pattern.  This will also be useful if, say, we want to add special classification here (something that @jnm2 wants) :)

Reply:
yes, I do want that, hard to remember which symbols need to be escaped otherwise